frontend/packages/dev-console/src/components/topology2/components/nodes/BaseNode.tsx

```diff
@@ -62,6 +65,15 @@ const BaseNode: React.FC<BaseNodeProps> = ({
   useAnchor(EllipseAnchor);
   const cx = element.getBounds().width / 2;
   const cy = element.getBounds().height / 2;
+  const resourceObj = getTopologyResourceObject(element.getData());
+  const resourceModel = modelFor(referenceFor(resourceObj));
+  const editAccess = useAccessReview({
+    group: resourceModel.apiGroup,
+    verb: 'patch',
+    resource: resourceModel.plural,
+    name: resourceObj.metadata.name,
+    namespace: resourceObj.metadata.namespace,
+  });
```

Comment:
Does the HOC and this hook cause 2 network calls?
If so we should be doing only one of these.

Reply:
No, both cause only one network call.

Reply:
It is making only one network call per node to check for the permission.

Reply:
I see that the request is memoized so it results only in a single network call.
We can consolidate the requests when we move away from HOC and use hooks.